GoogleDataTransport/GDTCORLibrary/GDTCORPlatform.m

```diff
@@ -330,30 +356,97 @@ - (instancetype)init {
   return self;
 }
 
+/** Populates a unique NSUInteger as GDTCORBackgroundIdentifier for given semaphore and maps it to
+ * the semaphore in NSMutableDictionary.
+ *
+ * @param semaphore The task semaphore needs to be mapping.
+ * @return A unique GDTCORBackgroundIdentifier mapped to the given semaphore.
+ */
++ (GDTCORBackgroundIdentifier)createAndMapBackgroundIdentifierToSemaphore:
+    (dispatch_semaphore_t)semaphore {
+  __block GDTCORBackgroundIdentifier bgID = GDTCORBackgroundIdentifierInvalid;
+  if (sSemaphoreQueue && sBackgroundIdentifierToSemaphoreMap) {
+    dispatch_sync(sSemaphoreQueue, ^{
+      bgID = arc4random();
+      while (bgID == GDTCORBackgroundIdentifierInvalid ||
+             sBackgroundIdentifierToSemaphoreMap[[NSNumber numberWithLong:bgID]]) {
+        bgID = arc4random();
+      }
+      sBackgroundIdentifierToSemaphoreMap[[NSNumber numberWithLong:bgID]] = semaphore;
```

Comment:
replace `[NSNumber numberWithLong:]` with just `@(bgID)`, here and several other places

Reply:
Fixed